Incorrectly merging built-in and user config `languages.toml` files for typescript lsp

<!-- Your issue may already be reported!
Please search on the issue tracker before creating one. -->

### Reproduction steps

You need to have [`deno`](https://deno.land/#installation) installed to be able to reproduce and consult wiki for helix config/log directories.

1. Create an empty `languages.toml` in the Helix config directory.
2. Copy in the text from [`helix/languages.toml`](https://github.com/helix-editor/helix/blob/master/languages.toml)
3. Navigate to the `language` entry that has a name of `typescript`
4. Change the `language-server` value to be of `{ command = "deno", args = ["lsp"] }` and save
5. Create a dummy file with a `ts` file extension
6. Insert the following code:

```ts
console.log("Hello, world!");
```

7. Close helix
8. Open the file in helix again with the `-v` flag
9. Open the helix log file and see the errors listed below.

### Environment

- Platform: Windows
- Helix version: 0.5

<details><summary><code>C:\Users\carte\AppData\Local\helix\helix.log</code></summary>

```
2021-11-06T21:20:10.786 helix_lsp::transport [INFO] -> {"jsonrpc":"2.0","method":"initialize","params":{"capabilities":{"textDocument":{"codeAction":{"codeActionLiteralSupport":{"codeActionKind":{"valueSet":["","quickfix","refactor","refactor.extract","refactor.inline","refactor.rewrite","source","source.organizeImports"]}}},"completion":{"completionItem":{"snippetSupport":false},"completionItemKind":{}},"hover":{"contentFormat":["markdown"]}},"window":{"workDoneProgress":true}},"processId":20380,"rootUri":null},"id":0}
2021-11-06T21:20:10.870 helix_lsp::transport [ERROR] err <- "\u{1b}[0m\u{1b}[1m\u{1b}[31merror\u{1b}[0m: Found argument '--stdio' which wasn't expected, or isn't valid in this context\n"
2021-11-06T21:20:10.870 helix_lsp::transport [ERROR] err <- "\n"
2021-11-06T21:20:10.870 helix_lsp::transport [ERROR] err <- "USAGE:\n"
2021-11-06T21:20:10.870 helix_lsp::transport [ERROR] err <- "    deno [OPTIONS] [SUBCOMMAND]\n"
2021-11-06T21:20:10.870 helix_lsp::transport [ERROR] err <- "\n"
2021-11-06T21:20:10.870 helix_lsp::transport [ERROR] err <- "For more information try --help\n"
2021-11-06T21:20:10.870 helix_lsp::transport [ERROR] err <- "\n"
2021-11-06T21:20:10.876 helix_lsp::transport [ERROR] err: <- StreamClosed
2021-11-06T21:20:10.876 helix_lsp::transport [ERROR] err: <- StreamClosed
```

</details>

--- a/helix-core/src/lib.rs
+++ b/helix-core/src/lib.rs
@@ -103,25 +103,7 @@ pub fn cache_dir() -> std::path::PathBuf {
 pub fn merge_toml_values(left: toml::Value, right: toml::Value) -> toml::Value {
     use toml::Value;
 
-    fn get_name(v: &Value) -> Option<&str> {
-        v.get("name").and_then(Value::as_str)
-    }
-
     match (left, right) {
-        (Value::Array(mut left_items), Value::Array(right_items)) => {
-            left_items.reserve(right_items.len());
-            for rvalue in right_items {
-                let lvalue = get_name(&rvalue)
-                    .and_then(|rname| left_items.iter().position(|v| get_name(v) == Some(rname)))
-                    .map(|lpos| left_items.remove(lpos));
-                let mvalue = match lvalue {
-                    Some(lvalue) => merge_toml_values(lvalue, rvalue),
-                    None => rvalue,
-                };
-                left_items.push(mvalue);
-            }
-            Value::Array(left_items)
-        }
         (Value::Table(mut left_map), Value::Table(right_map)) => {
             for (rname, rvalue) in right_map {
                 match left_map.remove(&rname) {
@@ -149,12 +131,13 @@ mod merge_toml_tests {
     fn language_tomls() {
         use toml::Value;
 
-        const USER: &str = "
+        const USER: &str = r#"
         [[language]]
-        name = \"nix\"
-        test = \"bbb\"
-        indent = { tab-width = 4, unit = \"    \", test = \"aaa\" }
-        ";
+        name = "typescript"
+        test = "bbb"
+        indent = { tab-width = 4, unit = "    ", test = "aaa" }
+        language-server = { command = "deno", args = ["lsp"] }
+        "#;
 
         let base: Value = toml::from_slice(include_bytes!("../../languages.toml"))
             .expect("Couldn't parse built-in langauges config");
@@ -162,23 +145,27 @@ mod merge_toml_tests {
 
         let merged = merge_toml_values(base, user);
         let languages = merged.get("language").unwrap().as_array().unwrap();
-        let nix = languages
+        let ts = languages
             .iter()
-            .find(|v| v.get("name").unwrap().as_str().unwrap() == "nix")
+            .find(|v| v.get("name").unwrap().as_str().unwrap() == "typescript")
             .unwrap();
-        let nix_indent = nix.get("indent").unwrap();
+        let ts_indent = ts.get("indent").unwrap();
 
         // We changed tab-width and unit in indent so check them if they are the new values
-        assert_eq!(
-            nix_indent.get("tab-width").unwrap().as_integer().unwrap(),
-            4
-        );
-        assert_eq!(nix_indent.get("unit").unwrap().as_str().unwrap(), "    ");
+        assert_eq!(ts_indent.get("tab-width").unwrap().as_integer().unwrap(), 4);
+        assert_eq!(ts_indent.get("unit").unwrap().as_str().unwrap(), "    ");
         // We added a new keys, so check them
-        assert_eq!(nix.get("test").unwrap().as_str().unwrap(), "bbb");
-        assert_eq!(nix_indent.get("test").unwrap().as_str().unwrap(), "aaa");
-        // We didn't change comment-token so it should be same
-        assert_eq!(nix.get("comment-token").unwrap().as_str().unwrap(), "#");
+        assert_eq!(ts.get("test").unwrap().as_str().unwrap(), "bbb");
+        assert_eq!(ts_indent.get("test").unwrap().as_str().unwrap(), "aaa");
+        assert_eq!(
+            ts.get("language-server")
+                .unwrap()
+                .get("args")
+                .unwrap()
+                .as_array()
+                .unwrap(),
+            &vec![Value::String("lsp".into())],
+        );
     }
 }
 

